crates/rolldown/src/utils/parse_to_ecma_ast.rs

```diff
@@ -83,7 +83,7 @@ pub fn parse_to_ecma_ast(
       (binary_to_esm(&encoded, options.platform, RUNTIME_MODULE_ID), OxcParseType::Js)
     }
     ModuleType::Asset => {
-      let content = "export default 'TODO: should be file path'".to_string();
+      let content = "export default import.meta.__ROLLDOWN_ASSET_FILENAME".to_string();
```

Comment:
```diff
diff --git a/crates/rolldown/src/utils/parse_to_ecma_ast.rs b/crates/rolldown/src/utils/parse_to_ecma_ast.rs
index a583d3729..f83a6c834 100644
--- a/crates/rolldown/src/utils/parse_to_ecma_ast.rs
+++ b/crates/rolldown/src/utils/parse_to_ecma_ast.rs
@@ -83,7 +83,8 @@ pub fn parse_to_ecma_ast(
       (binary_to_esm(&encoded, options.platform, RUNTIME_MODULE_ID), OxcParseType::Js)
     }
     ModuleType::Asset => {
-      let content = "export default import.meta.__ROLLDOWN_ASSET_FILENAME".to_string();
+      has_lazy_export = true;
+      let content = "import.meta.__ROLLDOWN_ASSET_FILENAME".to_string();
       (content, OxcParseType::Js)
     }
     ModuleType::Empty => (String::new(), OxcParseType::Js),

```

Reply:
this should align with esbuild, https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.js%00import+filePath+from+%27.%2Ffoo.txt%27%0Aconst+file+%3D+require%28%27.%2Fbar.txt%27%29%0Aconsole.log%28%60file%3A+%60%2C+file%29%0Aexport+default+filePath%0A&b=%00foo.txt%00hello%2C+world%0A&b=%00bar.txt%00bar&o=%7B%0A++treeShaking%3A+false%2C%0A++bundle%3A+true%2C%0A++format%3A+%22esm%22%0A%7D